compiler/rustc_mir_build/src/thir/pattern/check_match.rs

```diff
@@ -430,7 +430,10 @@ impl<'p, 'tcx> MatchVisitor<'p, 'tcx> {
         }
 
         let scrut_ty = scrut.ty;
-        let report = analyze_match(&cx, &tarms, scrut_ty);
+        let Ok(report) = analyze_match(&cx, &tarms, scrut_ty).map_err(|err| self.error = Err(err))
+        else {
+            return;
+        };
```

Comment:
```suggestion
        let report = match analyze_match(&cx, &tarms, scrut_ty) {
            Ok(report) => report,
            Err(err) => {
                self.error = Err(err);
                return;
            }
        };
```

I don't really like mixing combinators + side effects + let else 😅